### PR TITLE
[Build] skip renaming files for release wheels pipeline

### PR DESCRIPTION
**Pull request by simon-mo**
Previously all the wheels are named to v1.0.0.dev but this is not correct for published wheels during my release process. This PR 
* preserve the previous behavior for nightly
* fix the naming for release tag

it would be useful to review the shell script

### CODE REVIEW

File: .buildkite/upload-wheels.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# Rename 'linux' to 'manylinux1' in the wheel filename
+new_wheel="${wheel/linux/manylinux1}"
+mv -- "$wheel" "$new_wheel"
+wheel="$new_wheel"
+
+# Extract the version from the wheel
+version=$(unzip -p "$wheel" '**/METADATA' | grep '^Version: ' | cut -d' ' -f2)
+echo "Version: $version"
+
+# If the version contains "dev", rename it to v1.0.0.dev for consistency
```
Comment (khluu) [inline .buildkite/upload-wheels.sh:26]: I thought all wheels are named with `vllm-1.0.0.dev-cp38-abi3-manylinux1`. We don't have the count + commit hash in the name anymore. What other names are you seeing?

Comment (simon-mo) [reply .buildkite/upload-wheels.sh:26]: If you check the recent build, they are named by version+hash actually, it is just renamed to 1.0.0.dev

**Comment by comaniac** [comment]
Do you need to update the installation doc for the nightly wheel name as well?

**Comment by youkaichao** [comment]
what's the changing part?

**Comment by simon-mo** [comment]
This PR doesn't change the nightly behavior (renaming all wheels to 1.0.0.dev), it skips the nightly upload on main branch commit with the proper version tag so I can upload it to PyPI

**Pull request merged**